src/services/member/plugins/profile/repository.ts

```diff
@@ -7,7 +8,17 @@ import { MemberNotFound } from '../../../../utils/errors';
 import { MemberProfile } from './entities/profile';
 import { IMemberProfile } from './types';
 
-const MemberProfileRepository = AppDataSource.getRepository(MemberProfile).extend({
+class MemberProfileRepository {
+  private repository: Repository<MemberProfile>;
+
+  constructor(manager?: EntityManager) {
+    if (manager) {
+      this.repository = manager.getRepository(MemberProfile);
+    } else {
+      this.repository = AppDataSource.getRepository(MemberProfile);
+    }
+  }
```

Comment:
I think it can be replaced by `extends AbstractRepository<TYPE>`, check `MemberRepository`.